constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java

```diff
@@ -574,6 +596,36 @@ protected HttpUrl makeUrl(String path, UserInfo info) throws UnsupportedEncoding
         return url;
     }
 
+    /**
+     * Creates a builder for an authorized request
+     * 
+     * @return Request Builder
+     */
+    protected Builder makeAuthorizedRequestBuilder() {
+        Builder builder = new Request.Builder();
+        builder.addHeader("Authorization", this.credentials);
+        return builder;
+    }
+
+    /**
+     * Creates a builder for an end user request
+     * 
+     * @param info user information if available
+     * @return Request Builder
+     */
+    protected Builder makeUserRequestBuilder(UserInfo info) {
+      Builder builder = new Request.Builder();
+      if (info != null) {
+        if (this.constructorToken != null) {
```

Comment:
With this code `x-cnstrc-token` won't be set even if info is null. Is this intentional?

Reply:
Well played